deps/v8/src/objects.cc

```diff
@@ -7330,7 +7330,13 @@ namespace {
 
 Maybe<bool> GetPropertyDescriptorWithInterceptor(LookupIterator* it,
                                                  PropertyDescriptor* desc) {
-  if (it->state() == LookupIterator::INTERCEPTOR) {
+  bool has_access = true;
+  if (it->state() == LookupIterator::ACCESS_CHECK) {
+    has_access = it->HasAccess() || JSObject::AllCanRead(it);
+    it->Next();
```

Comment:
The access check is on the `object_template`, which is created by the embedder. We cannot configure templates as JSProxies. I'd say it's sound.